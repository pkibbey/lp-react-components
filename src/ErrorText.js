import React from 'react';
import { Text } from 'rebass';

const ErrorText = ({ name, error }) => (error.isError && error.hasInteracted
  ? (
    <Text
      variant="errorMessage"
      mb={3}
      mt={name === 'recaptcha' ? 2 : 0}
    >
      {error.message}
    </Text>
  )
  : null);
export default ErrorText;
