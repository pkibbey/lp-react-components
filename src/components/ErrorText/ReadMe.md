```js
<ErrorText error={{ isError: true, hasInteracted: true, message: 'This is an error' }} />
```
