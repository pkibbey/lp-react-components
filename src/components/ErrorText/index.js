import React from 'react'
import { Text } from 'rebass'
import theme from '../../theme'
import PropTypes from 'prop-types'

/**
 * Used to render an error message.
 *
 * @visibleName Error Text
 */
const ErrorText = ({ error, ...otherProps }) =>
  error.isError && error.hasInteracted ? (
    <Text
      sx={{
        fontSize: theme.fontSizes[0],
        fontFamily: '"Roboto", sans-serif',
        lineHeight: '20px',
        color: theme.colors.red,
        WebkitFontSmoothing: 'antialiased',
        maxWidth: ['100%', '100%', 414]
      }}
      {...otherProps}
    >
      {error.message}
    </Text>
  ) : null

ErrorText.propTypes = {
  /** An object describing the error */
  error: PropTypes.shape({
    isError: PropTypes.bool,
    hasInteracted: PropTypes.bool,
    message: PropTypes.string
  })
}

ErrorText.defaultProps = {
  error: {}
}

export default ErrorText
