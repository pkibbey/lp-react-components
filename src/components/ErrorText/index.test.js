import React from 'react'
import renderer from 'react-test-renderer'
import ErrorText from './'

it('renders error text when there is no error', () => {
  const tree = renderer
    .create(<ErrorText error={{ isError: false, message: 'test' }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders error text when there is an error', () => {
  const tree = renderer
    .create(
      <ErrorText
        error={{ isError: true, hasInteracted: true, message: 'test' }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders recaptcha error text when there is an error', () => {
  const tree = renderer
    .create(<ErrorText error={{ isError: true, message: 'test' }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
