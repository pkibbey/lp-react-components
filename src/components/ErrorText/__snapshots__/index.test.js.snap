// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error text when there is an error 1`] = `
<div
  className="css-14ff3hs"
>
  test
</div>
`;

exports[`renders error text when there is no error 1`] = `null`;

exports[`renders recaptcha error text when there is an error 1`] = `null`;
